packages/components/src/settings-form/index.tsx

```diff
@@ -13,7 +13,23 @@ import {
 	SettingPassword,
 	SettingCheckbox,
 	SettingSelect,
-} from './types';
+} from './setting-types';
+
+import { StringToString, Field, GetInputPropsReturn } from './types';
+
+interface SettingsFormProps {
+	fields: Field[] | { [ key: string ]: Field };
+	validate: ( values: StringToString ) => StringToString;
+	isBusy?: boolean;
+	onSubmit?: ( values: StringToString | void ) => void;
```

Comment:
Should this just be `( values?: Record<string, string> ) => void`?

or did you intend to cover some weird situations like passing undefined?, e.g. I guess you could do the following in TS

```
const myFn = (optionalArg: string | void) => {
    return !! optionalArg;
}

// these are all valid
myFn();
myFn('hi');
myFn(undefined);
```

Reply:
Good question, you're mainly talking about the `| void` bit? Honestly  not sure, probably copy/pasted it from somewhere, as I do.

Reply:
So looking at the Form component, there probably are *some* edge cases where this could be empty, but because `Form` is not typed currently we can't get much assurance on that. So maybe this could just be `Record<string, string>`?